components/choices/index.js

```diff
@@ -105,6 +111,39 @@ const updateChoiceById = async (req, response) => {
       throw new Error("No balance at proposal level");
     }
 
+    const isVoted = await db_connect
+      .collection("Choices")
+      .find({
+        pollID: poll._id,
+        walletAddresses: { $elemMatch: { address: address } },
+      })
+      .toArray();
+
+    if (isVoted.length > 0) {
+      const oldVote = await db_connect.collection("Choices").findOne({
+        _id: ObjectId(isVoted[0].walletAddresses[0].choiceId),
+      });
+
+      const oldSignaturePayload = oldVote.walletAddresses[0].payloadBytes;
+      if (oldSignaturePayload) {
+        const oldSignatureDate =
+          getTimestampFromPayloadBytes(oldSignaturePayload);
+
+        if (payloadDate <= oldSignatureDate) {
+          throw new Error("Invalid Signature");
+        }
+      }
+    }
+
+    const cidLink = await uploadToIPFS(
+      getIPFSProofFromPayload(payloadBytes, signature)
+    );
+    if (!cidLink) {
+      throw new Error(
+        "Could not upload proof to IPFS, Vote was not registered. Please try again later"
+      );
+    }
+
     await Promise.all(
```

Comment:
We don't want some promises to fulfill if one fails. If users votes on multiple choices all of them should be updates or none of them. Otherwise it will become a false vote.